opus/application/test_api/test_cart_api.py

```diff
@@ -2373,22 +2373,97 @@ def test__api_cart_datacsv_multi(self):
         self._run_json_equal(url, expected)
 
 
-            ################################################
-            ######### /__cart/view.html: API TESTS #########
-            ################################################
-
-    # XXX Need to implement tests
-
-
             ####################################################
             ######### /__cart/download.json: API TESTS #########
             ####################################################
+    def test__api_cart_download_single_no_hierarchical(self):
+        "[test_cart_api.py] /__cart/download.json: single opus id  & no hierarchical"
+        url = '/__cart/add.json?opusid=co-iss-n1462840881&reqno=456'
+        expected = {'recycled_count': 0, 'count': 1, 'error': False, 'reqno': 456}
+        self._run_json_equal(url, expected)
+        url = '/__cart/download.json?types=coiss_raw,coiss_calib,browse_full&hierarchical=0'
+        expected = ['N1462840881_1.IMG', 'N1462840881_1.LBL', 'prefix2.fmt', 'tlmtab.fmt', 'N1462840881_1_CALIB.IMG', 'N1462840881_1_CALIB.LBL', 'N1462840881_1_full.png', 'manifest.csv', 'data.csv', 'urls.txt']
+        self._run_zip_equal(url, expected)
```

Comment:
All of these tests leave .zip files behind in the TAR_FILE_PATH. They need to clean up after themselves by removing the zip file.